cbmc_viewer/optionst.py

```diff
@@ -518,12 +473,11 @@ def defaults(args):
     args = default_logging(args)
     args = handle_deprecated_arguments(args)
     args = default_source_method(args)
-    args = default_tags_method(args)
     warn_against_using_text_for_cbmc_output(args)
 
-    if hasattr(args, 'srcdir'):
+    if hasattr(args, 'srcdir') and args.srcdir is not None:
```

Comment:
IMHO it's less brittle to do stuff like this by using the [`type`](https://docs.python.org/3/library/argparse.html#type) argument to `add_argument`, when you're creating the parser. The simplest thing would be
```
parser.add_argument("--srcdir", type=os.path.abspath)
```
but you might even like to let the argument parser do some validation so that you fail as early as possible
```
def existing_directory(path):
    resolved = os.path.abspath(path)
    if not os.path.exists(resolved):
       raise argparse.ValueError("doesn't exist")
    if not os.path.isdir(resolved):
       raise argparse.ValueError("not a directory")
    return resolved

# ...

parser.add_argument("--srcdir", type=existing_directory)
```
and of course you can reuse the one function for several of these arguments.